torch/nn/modules/conv.py

```diff
@@ -682,8 +685,10 @@ def forward(self, input: Tensor, output_size: Optional[List[int]] = None) -> Ten
         if self.padding_mode != 'zeros':
             raise ValueError('Only `zeros` padding mode is supported for ConvTranspose1d')
 
+        # One cannot replace List by Sequence in "_output_padding" because pytorch does
+        # not support Sequence[T]. Therefore, a "type:ignore[...]" is required.
```

Comment:
With "pytorch does not support" I think you mean that TorchScript doesn't support it, right? It would be useful to document this `List`/`Tuple`/`Sequence` stuff (maybe on the wiki), because it will come back a lot. 

- Here all these things actually are tuples (because they're coerced to that with `_single`, `_double`, etc.). and `Tuple` is supported by torchscript. But everything is still annotated as `List[int]`, and the JIT seems happy with that. Unclear to me why it shouldn't be annotated as `Tuple[int, ...]` instead. I tried that on this PR, and it seems to not fail the tests in `test/jit/test_models.py` (however, it doesn't fail for `Sequence[int]` either, which was unexpected for me).
- `Conv*D.__init__` uses `_size_1_t` etc. annotations, which are unions. The JIT seems happy with that as well, it's unclear to me why (`Union` is not supported).
- It'd be useful to add `Sequence` to https://pytorch.org/docs/master/jit_language_reference.html#unsupported-typing-constructs
- Can we rely on tests breaking if we change one of these annotations in a way that would break torchscript? I assume yes, but if not we need a test/check procedure (e.g. check if feature is in https://pytorch.org/docs/master/jit_builtin_functions.html#supported-pytorch-functions).

@suo, @ezyang could you comment on this?

Reply:
I've added `Sequence` to the list of unsupported typing constructs

Reply:
actually I am not sure if this is necessary --> I guess the reason why you want to use Sequence here is bc _output_padding returns a List[int] but others expect a Tuple[int, ...]. but in fact it is a helper function only for this module. we can change its return .

also if that's the case you can also change the argument types to Tuple[int, ...] as well. ??

Reply:
Sorry for the late reply, but yes, problem with `Tuple[int, ...]` is that `torch.jit.script` can not parse that correctly. 
I've tried to fix that in https://github.com/pytorch/pytorch/pull/44774 but it requires a bit more work.
`_size_N_t` is a special type which is defined as `Union[int, Tuple[int * N]]`, which gets automagically promoted to list by torch.jit runtime.